edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,37 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let (lock, limit) = if let ConcurrencyThrottling::Limited(limit) = concurrency {
+            (Some(Arc::new(AtomicU32::new(*limit))), *limit)
+        } else {
+            (None, 0 as u32)
+        };
+
         let srv = incoming.for_each(move |(socket, addr)| {
-            let protocol = protocol.clone();
 
+            if let Some(lock) = lock.clone() {
+                let limit_reached = lock.fetch_update( Ordering::AcqRel,
```

Comment:
Might not work here, but I think there is a fetch_sub method on atomics that might be doing this logic here... Someone else might know better though :)

Reply:
I looked at it but I need a  special logic for failure